tests/kani/BoundedArbitrary/derive.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// kani-flags: --default-unwind 5
+//
+// Check that users can implement Arbitrary to a simple data struct with Vec<>.
+extern crate kani;
+use kani::{BoundedAny, BoundedArbitrary};
+
+#[derive(BoundedArbitrary)]
+#[allow(unused)]
+struct MyVector<T> {
+    #[bounded]
+    vector: Vec<T>,
+    cap: usize,
+}
+
+/// Check that macro works on enums
+#[derive(BoundedArbitrary)]
+#[allow(unused)]
+enum Enum<T> {
+    A(#[bounded] String),
+    B(#[bounded] Vec<T>, usize),
+    C {
+        #[bounded]
+        x: Vec<T>,
+        y: bool,
+    },
+}
+
+#[kani::proof]
+fn check_my_vec() {
+    let my_vec: BoundedAny<MyVector<bool>, 4> = kani::any();
+    assert!(my_vec.vector.len() <= 4)
```

Comment:
A stronger property is to ensure all vector lengths are possible, which you can do using:
```rust
kani::cover!(my_vec.vector.len() == 0);
kani::cover!(my_vec.vector.len() == 1);
kani::cover!(my_vec.vector.len() == 2);
kani::cover!(my_vec.vector.len() == 3);
kani::cover!(my_vec.vector.len() == 4);
```